sound/soc/sof/ipc4-pcm.c

```diff
@@ -893,11 +893,11 @@ static snd_pcm_sframes_t sof_ipc4_pcm_delay(struct snd_soc_component *component,
 	/* Calculate the delay taking into account that both pointer can wrap */
 	div64_u64_rem(tmp_ptr, substream->runtime->boundary, &tmp_ptr);
 	if (substream->stream == SNDRV_PCM_STREAM_PLAYBACK) {
-		head_ptr = substream->runtime->status->hw_ptr;
+		head_ptr = snd_sof_pcm_platform_pointer(sdev, substream);
 		tail_ptr = tmp_ptr;
 	} else {
 		head_ptr = tmp_ptr;
-		tail_ptr = substream->runtime->status->hw_ptr;
+		tail_ptr = snd_sof_pcm_platform_pointer(sdev, substream);
```

Comment:
commit title: ASoC: SOF: ipc4-pcm: ...

Reply:
and it's not clear to the average reviewer what snd_sof_pcm_platform_pointer() might do either or what this improves.

Reply:
use snd_sof_pcm_platform_pointer() to get latest host dma position.  Updated commit message, thanks for review